<launch>
    <arg name="detector_yaml" default="$(find perception_test)/config/detector.yaml" />
    <arg name="roi_lanes_yaml" default="$(find perception_test)/config/roi_lanes.yaml" />
    <arg name="offline_sources_json" default="$(find perception_test)/config/offline_sources.json" />
    <arg name="offline_source_name" default="default" />
    
    <node pkg="perception_test" type="offline_provider_node.py" name="offline_provider" output="screen">
        <param name="sources_json" value="$(arg offline_sources_json)" />
        <param name="source_name" value="$(arg offline_source_name)" />
    </node>

    <node pkg="perception_test" type="object_detector_node.py" name="object_detector" output="screen">
        <rosparam command="load" file="$(arg detector_yaml)" />
        <param name="topics/image_in" value="/camera/image_raw" />
        <param name="topics/detections_out" value="/perception_test/detections_raw" />
    </node>

    <node pkg="perception_test" type="lane_detector_node.py" name="lane_detector" output="screen">
        <rosparam command="load" file="$(arg roi_lanes_yaml)" />
        <param name="input_topic" value="/perception_test/detections_raw" />
        <param name="output_topic" value="/perception_test/detections" />
        <param name="roi_source/path" value="$(find perception_test)/src/nodes/lane_polys.json" />
    </node>

    <node pkg="perception_test" type="tracker_node.py" name="tracker" output="screen">
        <param name="in_topic" value="/perception_test/detections" />
        <param name="out_topic" value="/perception_test/tracks" />
        <param name="iou_threshold" value="0.3" />
    </node>

    <node pkg="perception_test" type="association_node.py" name="association" output="screen">
        <param name="radar/pointcloud_topic" value="/point_cloud" />
        <param name="bbox_topic" value="/perception_test/tracks" />
        <param name="associated_topic" value="/perception_test/associated" />
        <param name="extrinsic_path" value="$(find perception_test)/config/extrinsic.json" />
    </node>

    <node pkg="perception_test" type="speed_estimator_node.py" name="speed_estimator" output="screen">
        <param name="in_topic" value="/perception_test/associated" />
        <param name="out_topic" value="/perception_test/output" />
        <param name="gate/min_speed_kph" value="0.0" />
    </node>

    <node pkg="perception_test" type="calibration_in_node.py" name="cam_intrinsic_calib" output="screen">
    <param name="camera_topic" value="/camera/image_raw"/>
    <param name="board/width" value="7"/>
    <param name="board/height" value="5"/>
    <param name="board/square_size" value="0.10"/>
    <param name="min_samples" value="30"/>
    <param name="output_path" value="$(find perception_test)/config/camera_intrinsic.yaml"/>
    <param name="save_images/enable" value="true"/>
    <param name="save_images/path" value="$(find perception_test)/calibration_images"/>
    </node>

    <node pkg="perception_test" type="calibration_diagnosis_node.py" name="trajectory_eval" output="screen" />

</launch>